curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60f5d2776c854e069560fbe6.md

```diff
@@ -11,18 +11,39 @@ The first `fieldset` will hold name, email, and password fields. Start by adding
 
 # --hints--
 
-You should add four `label` elements.
+There should be three `fieldset` elements in total.
 
 ```js
-assert.equal(document.querySelectorAll('label')?.length, 4);
+assert.equal(document.querySelectorAll('fieldset')?.length, 3);
 ```
 
-You should add the `label` elements as direct children to the first `fieldset`. Make sure to close the `label` elements.
+The `fieldset` elements should all be direct children of the `form` element.
+
+```js
+assert.equal(document.querySelectorAll('form > fieldset')?.length, 3);
+```
+
+The four `label` elements should all be direct children of the first `fieldset` element. Make sure you close the `label` elements.
 
 ```js
 assert.equal(document.querySelectorAll('form fieldset:first-child > label')?.length, 4);
 ```
 
+The last two `fieldset` elements should be empty.
+
+```js
+const LastTwoFieldsetElements = Array.from(document.querySelectorAll("form fieldset:not(fieldset:first-child)"));
+assert.isTrue(LastTwoFieldsetElements.filter((ele) => ele?.innerHTML?.replace(/\s*/g, ""))?.length === 0);
+```
+
+Make sure you close the `label` elements.
+
+```js
+assert.lengthOf(document.querySelector('fieldset')?.innerHTML?.match(/<\/label>/g), 4)
```

Comment:
```suggestion
assert.lengthOf(document.querySelector('fieldset')?.innerHTML?.match(/<\/label\s*>/g), 4)
```

Reply:
Why do we need `\s*` here? I'm fine with using it I'm just not sure I understand why it is needed.

Reply:
Based on the spec: https://html.spec.whatwg.org/#syntax-end-tag
> After the tag name, there may be one or more [ASCII whitespace](https://infra.spec.whatwg.org/#ascii-whitespace).

Reply:
Right, but the spaces inside the tags are not part of the `innerHTML` (spaces inside the element are).
```
<fieldset>
  <label></label   >
</fieldset>
```
```
console.log(document.querySelector('fieldset')?.innerHTML) // <label></label>
```

Reply:
That is interesting. Thank you 👍